Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -172,26 +190,24 @@ private func parseBytes(
         var padding = 0 // # padding '=' chars
         digits: while true {
             let digit = source[index]
-            let k = base64Values[Int(digit)]
-            if k >= 0 {
-                n <<= 6
-                n |= k
-                chars += 1
-                if chars == 4 {
-                    p[0] = UInt8(truncatingIfNeeded: n >> 16)
-                    p[1] = UInt8(truncatingIfNeeded: n >> 8)
-                    p[2] = UInt8(truncatingIfNeeded: n)
-                    p += 3
-                    chars = 0
-                    n = 0
-                }
-            } else {
+            var k = base64Values[Int(digit)]
+            if k < 0 {
                 switch digit {
                 case asciiDoubleQuote:
                     source.formIndex(after: &index)
                     break digits
+                case asciiBackslash:
+                    source.formIndex(after: &index)
+                    let escaped = source[index]
+                    switch escaped {
+                    case asciiForwardSlash:
+                        k = base64Values[Int(escaped)]
+                    default:
+                        throw JSONDecodingError.malformedString
+                    }
                 case asciiSpace:
-                    break
+                    source.formIndex(after: &index)
```

Comment:
Why this change? It appears to make spaces permissible, and ignored, at most locations in the middle of the base64 data?

Reply:
It is because of the reflow of the loop, this now does a the `continue digits` where before it just did a `break` and so it fell to the code below that did this work.

Reply:
This is based on the Google C++ implementation, which is tolerant of spaces.  On the other hand, [RFC4648 Section 3.3](https://tools.ietf.org/html/rfc4648#section-3.3) urges people to reject base64 text with spaces.  I've chosen to ignore spaces to better match Google's implementation, but this would be easy to change if we had a reason.

Reply:
I added a test to document this behavior.